### Please describe the problem.

Tried to drop files that exist on s3:
 git annex drop -J36 --in=ilya-s3

A few get dropped, and then it errors out with
git-annex: content is locked

(Apart from the issue here, maybe add an option to keep working in case of error with one file?)

### What version of git-annex are you using? On what operating system?
git-annex version: 6.20180926-gc906aaf
build flags: Assistant Webapp Pairing S3(multipartupload)(storageclasses) WebDAV Inotify ConcurrentOutput TorrentParser MagicMime Feed\
s Testsuite
dependency versions: aws-0.17.1 bloomfilter-2.0.1.0 cryptonite-0.23 DAV-1.3.1 feed-0.3.12.0 ghc-8.0.2 http-client-0.5.7.0 persistent-s\
qlite-2.6.2 torrent-10000.1.1 uuid-1.3.13 yesod-1.4.5
key/value backends: SHA256E SHA256 SHA512E SHA512 SHA224E SHA224 SHA384E SHA384 SHA3_256E SHA3_256 SHA3_512E SHA3_512 SHA3_224E SHA3_2\
24 SHA3_384E SHA3_384 SKEIN256E SKEIN256 SKEIN512E SKEIN512 BLAKE2B256E BLAKE2B256 BLAKE2B512E BLAKE2B512 BLAKE2B160E BLAKE2B160 BLAKE\
2B224E BLAKE2B224 BLAKE2B384E BLAKE2B384 BLAKE2S256E BLAKE2S256 BLAKE2S160E BLAKE2S160 BLAKE2S224E BLAKE2S224 BLAKE2SP256E BLAKE2SP256\
 BLAKE2SP224E BLAKE2SP224 SHA1E SHA1 MD5E MD5 WORM URL
remote types: git gcrypt p2p S3 bup directory rsync web bittorrent webdav adb tahoe glacier ddar hook external
operating system: linux x86_64
supported repository versions: 3 5 6
upgrade supported from repository versions: 0 1 2 3 4 5
local repository version: 5


(master_env_py27_v28) [01:05 PM /data/ilya-work]$ uname -a
Linux ip-172-31-87-156 4.14.72-68.55.amzn1.x86_64 #1 SMP Fri Sep 28 21:14:54 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

### Please provide any additional information below.

> Made drop not operate on the same key multiple times concurrently.
> 
> Also yeah, that kind of exception should not interrupt processing other
> files; the distinction git-annex has drawn between IO errors and other
> exceptions is not super useful unless a command for some reason wants to
> completely stop everything for some reason. I can't think of any cases
> where a command would want to do that; if it does turn out to be needed,
> a special type of exception could be thrown to force termination.
> So I changed it to catch all (non-async) errors.
> 
> [[fixed|done]] --[[Joey]]
